shared/src/main/scala/io/kaitai/struct/languages/RustCompiler.scala

```
@@ -1,612 +1,1391 @@
+    with EveryReadIsExpression
+    with FixedContentsUsingArrayByteLiteral
     with ObjectOrientedLanguage
-    with UpperCamelCaseClasses
     with SingleOutputFile
-    with AllocateIOLocalVar
+    with UpperCamelCaseClasses
     with UniversalFooter
-    with UniversalDoc
-    with FixedContentsUsingArrayByteLiteral
-    with EveryReadIsExpression {
+    with SwitchIfOps
+    with UniversalDoc {
 
   import RustCompiler._
 
+  override val translator: RustTranslator =
+    new RustTranslator(typeProvider, config)
+
   override def innerClasses = false
 
   override def innerEnums = false
 
-  override val translator: RustTranslator = new RustTranslator(typeProvider, config)
-
-  override def universalFooter: Unit = {
-    out.dec
-    out.puts("}")
-  }
-
-  override def outImports(topClass: ClassSpec) =
-    importList.toList.map((x) => s"use $x;").mkString("", "\n", "\n")
-
   override def indent: String = "    "
+
   override def outFileName(topClassName: String): String = s"$topClassName.rs"
 
+  override def outImports(topClass: ClassSpec): String =
+    importList.toList
+      .mkString("", "\n", "\n")
+
   override def fileHeader(topClassName: String): Unit = {
     outHeader.puts(s"// $headerComment")
     outHeader.puts
 
-    importList.add("std::option::Option")
-    importList.add("std::boxed::Box")
-    importList.add("std::io::Result")
-    importList.add("std::io::Cursor")
-    importList.add("std::vec::Vec")
-    importList.add("std::default::Default")
-    importList.add("kaitai_struct::KaitaiStream")
-    importList.add("kaitai_struct::KaitaiStruct")
+    outHeader.puts("#![allow(unused_imports)]")
+    outHeader.puts("#![allow(non_snake_case)]")
+    outHeader.puts("#![allow(non_camel_case_types)]")
+    outHeader.puts("#![allow(irrefutable_let_patterns)]")
+    outHeader.puts("#![allow(unused_comparisons)]")
+    outHeader.puts("#![allow(arithmetic_overflow)]")
+    outHeader.puts("#![allow(overflowing_literals)]")
```

These two suppressions hide legitimate errors, so I'll remove them.